File: dsa_algo/leetcode/ts/src/arrays_hash/1603-design_parking_system.ts
class ParkingSystem {
    spaces:number [] = [];
    constructor(big: number, medium: number, small: number) {
        this.spaces = [big, medium, small];
    }

    addCar(carType: number): boolean {
        if(this.spaces[carType - 1] > 0)
        {
            this.spaces[carType - 1]--;
            return true;
        }
        return false;
    }
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * var obj = new ParkingSystem(big, medium, small)
 * var param_1 = obj.addCar(carType)
 */

File: dsa_algo/leetcode/arrays/arrays_hash/1603-design_parking_system.cpp
class ParkingSystem {
public:
    std::vector<int> spaces;
    ParkingSystem(int big, int medium, int small) {
       spaces = {big, medium, small};
    }
    
    bool addCar(int carType) {
       if(spaces[carType - 1] > 0) 
       {
           spaces[carType - 1]--;
           return true;
       }
       return false;
    }
};

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * ParkingSystem* obj = new ParkingSystem(big, medium, small);
 * bool param_1 = obj->addCar(carType);
 */